### PR TITLE
common components

### PR DESCRIPTION
**Pull request by jrassa**
Adds quick-filter, quick-column-toggle, search-table, and system-alerts components to common.  Updates to admin components to use these.

### CODE REVIEW

**Comment by jrassa** [comment]
This also addresses #25

File: src/styles/_display.scss [hunk 1]
```diff
@@ -71,3 +71,18 @@
 		padding-left: 5px * $i;
 	}
 }
+
+//----------------------------------------
+// Positioning
+//----------------------------------------
+.pull-left {
```
Comment (brianghig) [inline src/styles/_display.scss:78]: @jrassa do we need these, or are they dupes from what bootstrap provides?

File: src/app/core/feedback/admin/admin-list-feedback.component.html [hunk 2]
```diff
@@ -2,40 +2,51 @@
...
-	</pageable-table>
+	<div class="row padding-top-5">
+		<div class="col-md-12">
+
+			<!-- Export -->
+			<div class="row">
+				<div class="col-md-12 form-group">
+					<div class="btn-group pull-right dropdown">
+						<button type="button" class="btn btn-sm btn-outline-secondary" (click)="export()">
+							<i class="fa fa-download"></i> Export
```
Comment (brianghig) [inline src/app/core/feedback/admin/admin-list-feedback.component.html]: Now that Font Awesome supports "fa" classes on "span" tags in order to be more semantically accurate, I think we should start moving to `<span>` vs. `<i>`. Sounds like a new ticket to me, though 😄

File: src/app/core/feedback/admin/admin-list-feedback.component.html [hunk 3]
```diff
@@ -2,40 +2,51 @@
...
+					</th>
+				</ng-template>
+
+				<ng-template named-template="table-row" let-feedback>
+					<td>{{ feedback?.creator?.name }}</td>
+					<td>{{ feedback?.creator?.email }}</td>
+					<td>{{ feedback?.type }}</td>
+					<td>{{ feedback?.body }}</td>
+					<td>{{ feedback?.url }}</td>
+					<td [tooltip]="feedback.created" container="body" placement="top">{{ feedback?.created | agoDate:false }}</td>
```
Comment (brianghig) [inline src/app/core/feedback/admin/admin-list-feedback.component.html]: Should this `feedback` object be optional for consistency with the rest of the displays? Or does the difference of a data binding cover the null case?

**Review by brianghig — CHANGES_REQUESTED** [review]
Only one minor comment about the `pull-left` and related CSS classes that might already be provided by Bootstrap.

Are there any tests available to cover this new functionality?

File: src/styles/_display.scss (hunk 1 above)
Comment (jrassa) [reply src/styles/_display.scss:78]: As far as I can tell, `pull-left` and `pull-right` aren't coming from bootstrap.  They are currently coming from font-awesome, but they are marked as deprecated and will be removed in v5.  It also just seemed improper to rely on a class from font-awesome for general styling.

File: src/app/core/feedback/admin/admin-list-feedback.component.html (hunk 3 above)
Comment (jrassa) [reply src/app/core/feedback/admin/admin-list-feedback.component.html]: fixed

File: src/styles/_display.scss (hunk 1 above)
Comment (reblace) [reply src/styles/_display.scss:78]: https://getbootstrap.com/docs/4.0/migration/#utilities

Bootstrap v4 moved to "float-left" and "float-right".
http://getbootstrap.com/docs/4.0/utilities/float/

Comment (brianghig) [reply src/styles/_display.scss:78]: Thanks @reblace. I added #30 to cover that migration.

**Pull request merged**